cachi2/core/config.py

```diff
@@ -1,24 +1,24 @@
-config = None
+from typing import Optional
+
+import typer
+import yaml
+from pydantic import BaseModel
 
+config = None
 
-class Config:
-    """
-    Singleton that provides default configuration for the Cachi2 process.
 
-    All values currently need to be changed in this file.
-    """
+class Config(BaseModel):
+    """Singleton that provides default configuration for the Cachi2 process."""
 
-    cachito_goproxy_url = "https://proxy.golang.org,direct"
-    cachito_default_environment_variables = {
+    goproxy_url: Optional[str] = "https://proxy.golang.org,direct"
+    default_environment_variables: Optional[dict] = {
         "gomod": {"GOSUMDB": {"value": "off", "kind": "literal"}},
     }
-    cachito_gomod_download_max_tries = 5
-    cachito_gomod_strict_vendor = True
-    cachito_subprocess_timeout = 3600
+    gomod_download_max_tries: Optional[int] = 5
+    gomod_strict_vendor: Optional[bool] = True
+    subprocess_timeout: Optional[int] = 3600
 
 
-# This function is kept to avoid changing the old code too much
-# It should be removed with the refactoring of the config object
```

Comment:
Nitpick: this function no longer refers to `worker_config`. I think we should drop the `worker` name here.